### PR TITLE
Allow crash reporters registration during app bootstrap

### PR DESCRIPTION
**Pull request by ChristophWurst**
### Before

This changes the crash reporters registration from
* Create a crash reporter instance
* Query the registry
* Pass the object to the registry

… when we might not even use it.

### After

It now just appends a string to an array. The service is only built if we really need it. Thus 🐨:rocket:

``\OCP\Support\CrashReport\IRegistry`` is obsolete, so I've deprecated it.

---

For https://github.com/ChristophWurst/nextcloud_sentry/pull/207, but needs https://github.com/nextcloud/server/pull/21456 to work properly.

### CODE REVIEW

**Comment by rullzer** [comment]
> Thus :koala: :rocket: 

I don't know what this means. But I'm on board!

**Comment by ChristophWurst** [comment]
> I don't know what this means. But I'm on board!

It's a very lazy but also very fast coala

File: lib/private/AppFramework/Bootstrap/RegistrationContext.php [hunk 1]
```diff
@@ -189,6 +208,23 @@ public function delegateCapabilityRegistrations(array $apps): void {
...
 
+	/**
+	 * @param App[] $apps
+	 */
+	public function delegateCrashReporterRegistrations(array $apps, Registry $registry): void {
+		foreach ($this->crashReporters as $registration) {
+			try {
+				$registry->registerLazy($registration['class']);
+			} catch (Throwable $e) {
+				$appId = $registration['appId'];
```
Comment (rullzer) [inline lib/private/AppFramework/Bootstrap/RegistrationContext.php:219]: I like the hardening here. pushing a string to an array.
But better save than sorry

**Review by rullzer — APPROVED** [review]

File: lib/private/AppFramework/Bootstrap/RegistrationContext.php (hunk 1 above)
Comment (ChristophWurst) [reply lib/private/AppFramework/Bootstrap/RegistrationContext.php:219]: You never know

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
php-cs is failing

**Comment by rullzer** [comment]
CI says no

**Comment by ChristophWurst** [comment]
Christoph says yes

**Comment by rullzer** [comment]
> Christoph says yes

:koala: :rocket:

**Comment by rullzer** [comment]
CI still says no...
https://drone.nextcloud.com/nextcloud/server/29862/10/4

:koala: :bomb:

**Comment by ChristophWurst** [comment]
Christoph says yes, again.

Turns out I did not check the registry tests and I even found an uncovered edge case while writing the tests :koala: :rocket:

**Comment by rullzer** [comment]
:koala: :balloon:
